packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -105,8 +112,16 @@ void main() {
 }
 ''');
 
+    int observatoryPort;
```

Comment:
Yeah, I didn't want to mess with reading the port back from the logs.  Seemed easier to push the port number in via a command-line parameter.